src/Capgemini.PowerApps.PackageDeployerTemplate/PackageTemplateBase.cs

```diff
@@ -22,193 +17,77 @@ public abstract class PackageTemplateBase : ImportExtension
-            foreach (var sla in defaultSlas)
-            {
-                defaultSlasQuery.AddAttributeValue("name", sla);
-            }
-            var retrieveMultipleResponse = this.CrmSvc.RetrieveMultiple(defaultSlasQuery);
-
-            foreach (var defaultSla in retrieveMultipleResponse.Entities)
-            {
-                defaultSla["isdefault"] = true;
-                this.CrmSvc.Update(defaultSla);
-            }
-        }
+            this.PackageLog.Log($"Initializing {nameof(PackageTemplateBase)} extension.");
 
-        private void DeactivateSdkMessageProcessingSteps(IEnumerable<string> sdkStepsToDeactivate)
-        {
-            if (sdkStepsToDeactivate is null || !sdkStepsToDeactivate.Any())
-            {
-                this.PackageLog.Log("No SDK steps to deactivate have been configured.");
-                return;
-            }
+            this.ConfigDataStorage = ConfigDataStorage.Load(this.ImportConfigFilePath);
 
-            var executeMultipleResponse = this.CrmSvc.SetRecordStateByAttribute("sdkmessageprocessingstep", 1, 2, "name", sdkStepsToDeactivate);
-            if (executeMultipleResponse.IsFaulted)
-            {
-                this.PackageLog.Log($"Error deactivating SDK Message Processing Steps.", TraceEventType.Error);
-                this.PackageLog.LogExecuteMultipleErrors(executeMultipleResponse);
-            }
-        }
+            var crmServiceAdapter = new CrmServiceAdapter(this.CrmSvc);
+            var logger = new TraceLoggerAdapter(this.PackageLog);
 
-        private void DeactivateProcesses(IEnumerable<string> processesToDeactivate)
-        {
-            if (processesToDeactivate is null || !processesToDeactivate.Any())
-            {
-                this.PackageLog.Log("No processes to deactivate have been configured.");
-                return;
-            }
+            this.DataImporterService = new DataImporterService(logger, crmServiceAdapter);
+            this.ProcessActivatorService = new ProcessActivatorService(logger, crmServiceAdapter);
+            this.SlaActivatorService = new SlaActivatorService(logger, crmServiceAdapter);
+            this.WordTemplateImporterService = new WordTemplateImporterService(logger, crmServiceAdapter);
+            this.SdkStepsActivatorService = new SdkStepsActivatorService(logger, crmServiceAdapter);
 
-            var executeMultipleResponse = this.CrmSvc.SetRecordStateByAttribute("workflow", 0, 1, "name", processesToDeactivate);
-            if (executeMultipleResponse.IsFaulted)
-            {
-                this.PackageLog.Log($"Error deactivating processes.", TraceEventType.Error);
-                this.PackageLog.LogExecuteMultipleErrors(executeMultipleResponse);
-            }
+            this.BeforeAnything();
         }
 
-        private void ActivateProcesses(IEnumerable<string> processesToActivate)
+        public void BeforeAnything()
```

Comment:
For consistency I think this ought to be `virtual` so it can be overridden in project templates similar to the rest of the methods.